### DISCUSSION TITLE
Ayuda con el Global State

### DISCUSSION THREAD

**Discussion by carmauc** [codigoencasa/builderbot#944] [Q&A]
Hola leifer, me puedes ayudar, lo que sucede es que tengo el siguiente codigo en el flow agente
![Snapshot_231230124612](https://github.com/codigoencasa/bot-whatsapp/assets/111944785/8f767f1a-5731-4d46-97eb-a00ab93abbc3)
Funciona correcto, cuando se marca la opcion se apaga el bot, sin embargo para los otros clientes que se comunican, el flow continua apagado.

Mi pregunta es si hay forma de apagar el bot solo para un cliente en especifico y que para los demas continue activo. Gracias

**Comment by JeanCaicedo** [comment]
El problema es que `globalState` es compartido entre **todas** las conversaciones — al apagar el bot ahi, lo apagas para todos. Tienes dos opciones para hacerlo por cliente:

**1. Usa `state` en vez de `globalState`** — el `state` es independiente por conversacion:

```js
.addAction(async (ctx, { state, endFlow }) => {
  if (state.get('botOff')) return endFlow()
})
```

y cuando el cliente elige apagarlo: `await state.update({ botOff: true })`

**2. Usa el `blacklist` integrado** — esta pensado justo para silenciar el bot a numeros especificos:

```js
.addAction(async (ctx, { blacklist, flowDynamic }) => {
  blacklist.add(ctx.from)   // el bot deja de responder solo a este numero
  await flowDynamic('Bot desactivado para ti')
  // blacklist.remove(ctx.from) para reactivarlo
})
```

Con cualquiera de las dos, los demas clientes siguen funcionando normal.
